api/application.go

```diff
@@ -142,17 +145,21 @@ func (a *ApplicationAPI) GetApplications(ctx *gin.Context) {
 //         $ref: "#/definitions/Error"
 func (a *ApplicationAPI) DeleteApplication(ctx *gin.Context) {
 	withID(ctx, "id", func(id uint) {
-		if app := a.DB.GetApplicationByID(id); app != nil && app.UserID == auth.GetUserID(ctx) {
-			if app.Internal {
-				ctx.AbortWithError(400, errors.New("cannot delete internal application"))
-				return
+		app, err := a.DB.GetApplicationByID(id)
+		if success := checkErrorOrAbort(ctx, 500, err); success {
```

Comment:
What do you think about returning early in case of an error via a db request.
This function could look like this:
```go
func (a *ApplicationAPI) DeleteApplication(ctx *gin.Context) {
	withID(ctx, "id", func(id uint) {
		app, err := a.DB.GetApplicationByID(id)
		if success := checkErrorOrAbort(ctx, 500, err); !success {
			return
		}
		if app == nil || app.UserID != auth.GetUserID(ctx) {
			ctx.AbortWithError(404, fmt.Errorf("app with id %d doesn't exists", id))
			return
		}

		if app.Internal {
			ctx.AbortWithError(400, errors.New("cannot delete internal application"))
			return
		}
		
		if success := checkErrorOrAbort(ctx, 500, a.DB.DeleteApplicationByID(id)); !success {
			return
		}
		
		if app.Image != "" {
			os.Remove(a.ImageDir + app.Image)
		}
	})
}
```
I'd say it is much more readable. It would be great if you could do this to the newly added ifs with chechErrorOrAbort, so the old code which doesn't follow return early can stay. (I could do it to, as it is not the most exiting thing todo (: )

Reply:
I wrote it that way so codecov is not going to complain about the coverage of that early return statement. If you are okay with not having 100% coverage I will use early return.

Reply:
Yeah, code quality / clarity > codecov (:.